### ISSUE TITLE
error in test phase

### ISSUE THREAD

**Issue by YoungMagic** [facebookresearch/Detectron#95]
### Expected results

I tried to train a model following the tutorial GETTING_STARTED.MD. I run the code bellow:

```
python2 tools/train_net.py \
    --cfg configs/getting_started/tutorial_1gpu_e2e_faster_rcnn_R-50-FPN.yaml \
    OUTPUT_DIR /tmp/detectron-output
```
The training process is throughly ok. Seems that it go wrong during computing mAP.

### Actual results

Please see the info.
```
INFO test_engine.py: 179: im_detect: range [1, 40504] of 40504: 40481/40504 0.056s + 0.002s (eta: 0:00:01)
INFO test_engine.py: 179: im_detect: range [1, 40504] of 40504: 40491/40504 0.056s + 0.002s (eta: 0:00:00)
INFO test_engine.py: 179: im_detect: range [1, 40504] of 40504: 40501/40504 0.056s + 0.002s (eta: 0:00:00)
INFO test_engine.py: 212: Wrote detections to: /tmp/detectron-output/test/coco_2014_minival/generalized_rcnn/detections.pkl
INFO test_engine.py:  64: Total inference time: 2932.182s
INFO task_evaluation.py:  75: Evaluating detections
INFO json_dataset_evaluator.py: 162: Writing bbox results json to: /tmp/detectron-output/test/coco_2014_minival/generalized_rcnn/bbox_coco_2014_minival_results.json
Loading and preparing results...     
DONE (t=14.27s)
creating index...
index created!
Traceback (most recent call last):
  File "tools/train_net.py", line 358, in <module>
    main()
  File "tools/train_net.py", line 199, in main
    test_model(checkpoints['final'], args.multi_gpu_testing, args.opts)
  File "tools/train_net.py", line 354, in test_model
    test_net.main(multi_gpu_testing=multi_gpu_testing)
  File "/home/mq/detectron/tools/test_net.py", line 128, in main
    results = parent_func(multi_gpu=multi_gpu_testing)
  File "/home/mq/detectron/lib/core/test_engine.py", line 66, in test_net_on_dataset
    dataset, all_boxes, all_segms, all_keyps, output_dir
  File "/home/mq/detectron/lib/datasets/task_evaluation.py", line 59, in evaluate_all
    dataset, all_boxes, output_dir, use_matlab=use_matlab
  File "/home/mq/detectron/lib/datasets/task_evaluation.py", line 79, in evaluate_boxes
    dataset, all_boxes, output_dir, use_salt=not_comp, cleanup=not_comp
  File "/home/mq/detectron/lib/datasets/json_dataset_evaluator.py", line 138, in evaluate_boxes
    coco_eval = _do_detection_eval(json_dataset, res_file, output_dir)
  File "/home/mq/detectron/lib/datasets/json_dataset_evaluator.py", line 193, in _do_detection_eval
    coco_eval = COCOeval(json_dataset.COCO, coco_dt, 'bbox')
TypeError: __init__() takes at most 3 arguments (4 given)
```

### System information

* Operating system: Ubuntu 16.04
* Compiler version: Cmake
* CUDA version: CUDA 8.0
* cuDNN version: cuDNN 5.1
* NVIDIA driver version: 384.111
* GPU models (for all devices if they are not all the same): GTX1080
* `PYTHONPATH` environment variable: /home/mq/detectron/lib:/home/mq/caffe2/build:/usr/local:/opt/py-faster-rcnn-master/lib:/opt/py-faster-rcnn-master/caffe-fast-rcnn/python: 
xargs: /home/mq/detectron/lib:/home/mq/caffe2/build:/usr/local:/opt/py-faster-rcnn-master/lib:
* `python --version` output: Python 2.7.12
* Dataset: COCO2014

Anyone can help?

**Comment by rbgirshick** [comment]
You have an old version of the COCO API. Please upgrade to the current master from https://github.com/cocodataset/cocoapi/.

**Comment by YoungMagic** [comment]
@rbgirshick Thanks for the help. But I just installed the COCO API several days ago. I happened to find that it's the json file that caused the problem. I downloaed the COCO2014 json file from the COCO official website. And to run the script I changed the val dataset from `instances_val2014.json` to `instances_minival2014.json`. Then it caused the problem. After I downloaded the  `instances_minival2014.json` from `(https://s3-us-west-2.amazonaws.com/detectron/coco/coco_annotations_minival.tgz)`, and replaced with it. The test process is fine. But i still don't know why it went wrong before. It should be fine even I changed the json file. Sorry to bother!
